utils/common/src/main/java/org/apache/brooklyn/util/collections/CollectionFunctionals.java

```diff
@@ -107,12 +107,12 @@ public Integer apply(Iterable<?> input) {
     }
 
     public static final class FirstElementFunction<T> implements Function<Iterable<? extends T>, T> {
-        private FirstElementFunction() {
+        public FirstElementFunction() {
```

Comment:
is this needed?  we have a static method `CollectionFunctionals.firstElement()` to return the instance, and the class is `final` so you can't subclass so i'm not sure what benefit there is from making this public.  (a javadoc on the constructor pointing at the method could be useful though!)

no strong feelings though